tests/foreman/api/test_discoveryrule.py

```diff
@@ -24,91 +24,95 @@
-        discovery_rule = entities.DiscoveryRule(
-            name=name,
-            search_=search,
-            hostname=hostname,
-            organization=[org],
-            location=[loc],
-            hostgroup=hostgroup,
-        ).create()
-        assert name == discovery_rule.name
-        assert hostname == discovery_rule.hostname
-        assert search == discovery_rule.search_
-        assert org.name == discovery_rule.organization[0].read().name
-        assert loc.name == discovery_rule.location[0].read().name
-        assert discovery_rule.enabled is True
-        # Update discovery rule
-        name = gen_choice(valid_data_list())
-        search = 'Location = Default_Location'
-        org = entities.Organization().create()
-        loc = entities.Location().create()
-        hostgroup = entities.HostGroup(organization=[org]).create()
-        max_count = gen_integer(1, 100)
-        enabled = False
-        discovery_rule.name = name
-        discovery_rule.search_ = search
-        discovery_rule.organization = [org]
-        discovery_rule.location = [loc]
-        discovery_rule.hostgroup = hostgroup
-        discovery_rule.max_count = max_count
-        discovery_rule.enabled = enabled
-        discovery_rule = discovery_rule.update(
-            ['name', 'organization', 'location', 'hostgroup', 'search_', 'max_count', 'enabled']
-        )
-        assert name == discovery_rule.name
-        assert search == discovery_rule.search_
-        assert org.name == discovery_rule.organization[0].read().name
-        assert loc.name == discovery_rule.location[0].read().name
-        assert hostgroup.id == discovery_rule.hostgroup.id
-        assert max_count == discovery_rule.max_count
-        assert enabled == discovery_rule.enabled
-        # Delete discovery rule
-        discovery_rule.delete()
-        with pytest.raises(HTTPError):
-            discovery_rule.read()
-
-    @tier1
-    def test_negative_create_with_invalid_host_limit_and_priority(self):
-        """Create a discovery rule with invalid host limit and priority
-
-        :id: e3c7acb1-ac56-496b-ac04-2a83f66ec290
-
-        :expectedresults: Validation error should be raised
-        """
-        with pytest.raises(HTTPError):
-            entities.DiscoveryRule(max_count=gen_string('alpha')).create()
-        with pytest.raises(HTTPError):
-            entities.DiscoveryRule(priority=gen_string('alpha')).create()
+
+
+@pytest.fixture(scope="module")
+def module_hostgroup(module_org):
+    module_hostgroup = entities.HostGroup(organization=[module_org]).create()
+    yield module_hostgroup
+    module_hostgroup.delete()
+
+
+@pytest.fixture(scope="module")
+def module_location(module_location):
+    yield module_location
+    module_location.delete()
+
+
+@pytest.fixture(scope="module")
+def module_org(module_org):
+    yield module_org
+    module_org.delete()
+
+
+@tier1
+def test_positive_end_to_end_crud(module_org, module_location, module_hostgroup):
```

Comment:
Module org is already defined entity_fixtures file commented above.

Reply:
@jyejare  Yes, they are defined in entity_fixure.py, but there's no clean up after creating those entities. Those fixtures I've defined in test_discoveryrule.py are basically a wrapper that uses them. Therefore even if the test DiscoveryRule fails, those org and location entities will be deleted properly. I wasn't sure if I should place them in entity_fixtures.py or modify those existing ones. Didn't want to break smth.